src/pdn/test/pdn_aux.py

```diff
@@ -0,0 +1,676 @@
+import re
+import utl
+import pdn
+import odb
+from collections import defaultdict
+
+
+#  In tcl land, this lives in OpenRoad.tcl. However, it seems to be only called
+#  in the pdn regression tests so it is defined here instead.
+def add_global_connection(design, *,
+                          net_name=None,
+                          inst_pattern=None,
+                          pin_pattern=None,
+                          power=False,
+                          ground=False,
+                          region=None):
+    if net_name is None:
+        utl.error(utl.PDN, 1501, "The net option for the " +
+                  "add_global_connection command is required.")
+
+    if inst_pattern is None:
+        inst_pattern = ".*"
+
+    if pin_pattern is None:
+        utl.error(utl.PDN, 1502, "The pin_pattern option for the " +
+                  "add_global_connection command is required.")
+
+    net = design.getBlock().findNet(net_name)
+    if net is None:
+        net = odb.dbNet_create(design.getBlock(), net_name)
+
+    if power and ground:
+        utl.error(utl.PDN, 1551, "Only power or ground can be specified")
+    elif power:
+        net.setSpecial()
+        net.setSigType("POWER")
+    elif ground:
+        net.setSpecial()
+        net.setSigType("GROUND")
+
+    # region = None
+    if region is not None:
+        region = design.getBlock().findRegion(region)
+        if region is None:
+            utl.error(utl.PDN, 1504, f"Region {region} not defined")
+
+    design.getBlock().addGlobalConnect(region, inst_pattern, pin_pattern, net, True)
+
+
+# Only to change the case? Seems weird.
```

Comment:
This was a backwards compatibility issue with the old PDNGEN

Reply:
OK

Reply:
I think we can drop backward compatibility now as everyone has moved on afaik.  If they haven't they are going to have bigger problems...